File: network-equipment-pms/pms-implementation/src/main/java/com/dp/plat/implementation/entity/TaskActivity.java
package com.dp.plat.implementation.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.Version;
import com.dp.plat.common.entity.BaseEntity;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * 任务活动记录实体（追加型），记录任务全生命周期事件。
 * 活动类型：CREATE/UPDATE/STATUS_CHANGE/SUBMIT_REVIEW/APPROVE/REJECT/
 * CHECKLIST_CHECK/COMMENT/PROGRESS_CHANGE/ASSIGN/MOVE。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("pms_task_activity")
public class TaskActivity extends BaseEntity {

    @NotNull(message = "任务ID不能为空")
    private Long taskId;

    @NotNull(message = "操作人ID不能为空")
    private Long userId;

    @Size(max = 64, message = "操作人姓名长度不能超过 64 个字符")
    private String userName;

    /** 活动类型。 */
    @NotBlank(message = "活动类型不能为空")
    @Size(max = 50, message = "活动类型长度不能超过 50 个字符")
    private String activityType;

    /** 活动描述。 */
    private String content;

    /** 附加元数据（JSON 字符串，如 old_value/new_value）。 */
    private String metadata;

    /** 乐观锁版本号。 */
    @Version
    private Integer version;
}

File: network-equipment-pms/pms-frontend/src/api/task-activity.ts
import { get } from '@/utils/request'

/**
 * 任务活动记录 API（只读）。对应后端 {@code TaskActivityController}，
 * 挂载在 {@code /api/implementation/task/activity} 下。
 *
 * <p>活动由各业务操作（创建/更新/状态变更/提交评审/审批/驳回/勾选/评论/
 * 进度变更/分配/移动）内部追加记录，前端只负责展示。</p>
 */

/** 任务活动记录 */
export interface TaskActivityItem {
  id?: number
  taskId: number
  userId?: number
  userName?: string
  /** 活动类型：CREATE/UPDATE/STATUS_CHANGE/SUBMIT_REVIEW/APPROVE/REJECT/... */
  activityType: string
  content?: string
  /** 附加元数据（JSON 字符串，如 old_value/new_value） */
  metadata?: string | null
  version?: number
  createTime?: string
  updateTime?: string
}

/** 查询任务活动记录（按时间倒序） */
export function listActivities(taskId: number): Promise<TaskActivityItem[]> {
  return get<TaskActivityItem[]>(`/api/implementation/task/activity/${taskId}`)
}
